extensions/http-curl/tests/HttpGetIntegrationTest.cpp

```diff
@@ -63,78 +62,134 @@ class HttpResponder : public CivetHandler {
+    controller_->waitUnload(60000);
+    LogTestController::getInstance().reset();
+  }
 
-  std::shared_ptr<core::Repository> test_repo = std::make_shared<TestRepository>();
-  std::shared_ptr<core::Repository> test_flow_repo = std::make_shared<TestFlowRepository>();
+  void run() {
+    controller_->start();
+    runAssertions();
+  }
 
-  configuration->set(minifi::Configure::nifi_flow_configuration_file, args.test_file);
+ protected:
+  virtual void runAssertions() {
+    assert(org::apache::nifi::minifi::utils::verifyLogLinePresenceInPollTime(
+        std::chrono::seconds(10),
+        "key:invokehttp.request.url value:" + url_,
+        "key:invokehttp.status.code value:200",
+        "key:flow.id"));
+  }
 
-  std::shared_ptr<minifi::io::StreamFactory> stream_factory = minifi::io::StreamFactory::getInstance(configuration);
+  void setupFlow() {
+    configuration_->set(minifi::Configure::nifi_default_directory, args_.key_dir);
+    configuration_->set(minifi::Configure::nifi_flow_configuration_file, args_.test_file);
+    content_repo_->initialize(configuration_);
 
-  std::shared_ptr<core::ContentRepository> content_repo = std::make_shared<core::repository::VolatileContentRepository>();
+    std::unique_ptr<core::FlowConfiguration> yaml_ptr = std::unique_ptr<core::YamlConfiguration>(
+        new core::YamlConfiguration(test_repo_, test_repo_, content_repo_, stream_factory_, configuration_, args_.test_file));
 
-  content_repo->initialize(configuration);
+    controller_ = std::make_shared<minifi::FlowController>(
+        test_repo_, test_flow_repo_, configuration_, std::move(yaml_ptr), content_repo_, DEFAULT_ROOT_GROUP_NAME, true);
+  }
 
-  std::unique_ptr<core::FlowConfiguration> yaml_ptr = std::unique_ptr<core::YamlConfiguration>(
-      new core::YamlConfiguration(test_repo, test_repo, content_repo, stream_factory, configuration, args.test_file));
-  std::shared_ptr<TestRepository> repo = std::static_pointer_cast<TestRepository>(test_repo);
+  void setupURL() {
+    core::YamlConfiguration yaml_config(test_repo_, test_repo_, content_repo_, stream_factory_, configuration_, args_.test_file);
 
-  std::shared_ptr<minifi::FlowController> controller = std::make_shared<minifi::FlowController>(
-      test_repo, test_flow_repo, configuration, std::move(yaml_ptr), content_repo, DEFAULT_ROOT_GROUP_NAME, true);
+    std::shared_ptr<core::Processor> proc = yaml_config.getRoot()->findProcessorByName("invoke");
+    assert(proc != nullptr);
 
-  core::YamlConfiguration yaml_config(test_repo, test_repo, content_repo, stream_factory, configuration, args.test_file);
+    const auto inv = std::dynamic_pointer_cast<minifi::processors::InvokeHTTP>(proc);
+    assert(inv != nullptr);
 
-  std::shared_ptr<core::Processor> proc = yaml_config.getRoot()->findProcessorByName("invoke");
-  assert(proc != nullptr);
+    inv->getProperty(minifi::processors::InvokeHTTP::URL.getName(), url_);
+  }
+
+  void setupServer(CivetHandler& http_handler) {
```

Comment:
there seems to be a similar class `VerifyRetryInvokeHTTP` in `VerifyInvokeHTTPTests.cpp`, how do the two relate?